### PR TITLE
Use the release version of `bazel-differ`

### PR DESCRIPTION
**Pull request by shs96c**
This simplifies our Go dependency management, and prevents `bazel-differ` from treading on dependencies that we need ourselves.

This change does not update the version of `bazel-differ` used. The only change is switching to a pre-built version of the binary rather than building it ourselves.

### CODE REVIEW

File: tests/integration/java/com/github/bazel_contrib/target_determinator/integration/BUILD.bazel [hunk 1]
```diff
@@ -120,3 +121,19 @@ java_library(
...
+    out = "bazel-differ",
+)
+
+alias(
+    name = "original-bazel-differ",
+    actual = select({
+        "//:linux_arm64": "@bazel_differ_linux_arm64//file",
+        "//:linux_x86_64": "@bazel_differ_linux_x86_64//file",
+        "//:macos_arm64": "@bazel_differ_darwin_arm64//file",
+        "//:macos_x86_64": "@bazel_differ_darwin_x86_64//file",
```
Comment (sitaktif) [inline tests/integration/java/com/github/bazel_contrib/target_determinator/integration/BUILD.bazel:137]: Please add support for windows as well as it's a supported platform (see e.g. https://github.com/bazel-contrib/target-determinator/blob/61d25f5e13a4b2205b338531b929464e3ed5eb3d/rules/multi_platform_go_binary.bzl#L8)

Comment (shs96c) [reply tests/integration/java/com/github/bazel_contrib/target_determinator/integration/BUILD.bazel:137]: While _we_ may support Windows, `bazel-differ` don't test with it, and they don't ship a binary for it, so it seems reasonable to assume that they don't support it.

File: third_party/go/deps.bzl [hunk 2]
```diff
@@ -97,6 +97,12 @@ def go_dependencies():
         sum = "h1:0VruCpn7yAIIu7pWVClQC8wxCJEcG3nyzpMSHKi1PQc=",
         version = "v1.6.0",
     )
+    go_repository(
+        name = "com_github_golang_protobuf",
+        importpath = "github.com/golang/protobuf",
+        sum = "h1:ROPKBNFfQgOUMifHyP+KYbvpjbdoFNs+aK7DXlji0Tw=",
+        version = "v1.5.2",
+    )
```
[the remark this answers is not in the document]
Comment (shs96c) [reply third_party/go/deps.bzl:105]: Previously, we had two `gazelle:repository_macro` entries, and were also pulling in deps from `bazel-differ`'s go_repositories file. This list is now purely ours, without any contamination from `bazel-diff`

[the remark this answers is not in the document]
Comment (gibfahn) [reply third_party/go/deps.bzl:105]: Okay, so this is the list of things that target-determinator and bazel-differ both rely on.

File: tests/integration/java/com/github/bazel_contrib/target_determinator/integration/BUILD.bazel (hunk 1 above)
Comment (sitaktif) [reply tests/integration/java/com/github/bazel_contrib/target_determinator/integration/BUILD.bazel:137]: Oh right, we're cross-building the windows binary from linux 👍

**Review by sitaktif — APPROVED** [review]

File: third_party/go/deps.bzl (hunk 2 above)
[the remark this answers is not in the document]
Comment (shs96c) [reply third_party/go/deps.bzl:105]: This is what get's generated by `scripts/update-dependencies`, so I believe that's right :)

**Pull request merged**